### ISSUE TITLE
[Bug] sign-app signature.json "certificate" field empty

### ISSUE THREAD

**Issue by Heiss** [owncloud/core#38779]
Cost me hours and hours to figure it out, that it is a version 10.7 problem. :)

### Steps to reproduce
1. Install ownCloud classic v10.7
2. run `occ integrity:sign-app  --privateKey="..." --certificate="..." --path="..."`
3. check `certificate` field in `signature.json`

### Expected behaviour
The given certificate should be printed in the `certificate` field in `signature.json`.

### Actual behaviour
`signature.json` holds the following value:
```
"certificate": "-----BEGIN CERTIFICATE-----\r\n\r\n-----END CERTIFICATE-----"
```

If i use the 10.6. installation, everything works like it should.

### Server configuration
**Operating system**: ubuntu20.04.1
**PHP version:** PHP 7.2.34-18+ubuntu20.04.1

**Updated from an older ownCloud or fresh install:** fresh install

**Where did you install ownCloud from:** From official webpage owncloud.com and docker image owncloud/server. Both the same issue with version 10.7.

**Comment by JammingBen** [comment]
Can confirm, this is probably related to the major upgrade of `phpseclib`. Thanks for the hint, will follow up on this!

**Comment by JammingBen** [comment]
Closing in favor of https://github.com/owncloud/enterprise/issues/4595

**Comment by Heiss** [comment]
@JammingBen Hey. Thank you for your support, but i cannot access the enterprise repo. Please write here, if the bug was fixed in a new release. :) Thank you and have a great weekend.

**Comment by JammingBen** [comment]
@Heiss Ahh okay, I'll keep you updated 👍

**Comment by JammingBen** [comment]
@Heiss Apparently the issue is related to your certificate version. Since OC 10.7, the version of a certificate should be 3. This change was not intended though, so we will come up with a fix for this. In the meantime you could maybe try with a v3 certificate?

A bit more information about versions: https://datatracker.ietf.org/doc/html/rfc5280#section-4.1.2.1

**Comment by JammingBen** [comment]
This is fixed with OC 10.8.0 which will be shipped at the beginning of July 2021.

**Comment by Heiss** [comment]
Thank you for your information. Then this can be closed. @JammingBen
